python-sdk/nuimages/scripts/render_rare_classes.py

```diff
@@ -0,0 +1,80 @@
+import argparse
+import random
+from collections import defaultdict
+from typing import Dict, Any, List
+
+from nuimages.nuimages import NuImages
+from nuimages.scripts.render_images import render_images
+
+
+def render_rare_classes(nuim: NuImages,
+                        render_args: Dict[str, Any],
+                        filter_categories: List[str] = None,
+                        max_frequency: float = 0.001) -> None:
+    """
+    Wrapper around render_images() that renders images with rare classes.
+    :param nuim: NuImages instance.
+    :param render_args: The render arguments passed on to the render function. See render_images().
+    :param filter_categories: Specify a list of object_ann category names.
+        Every sample that is rendered must contain annotations of any of those categories.
+        Filter_categories are a applied on top of the frequency filering.
+    :param max_frequency: The maximum relative frequency of the categories, at least one of which is required to be
+        present in the image. E.g. 0.1 indicates that one of the classes that account for at most 10% of the annotations
+        is present.
+    """
+    # Checks.
+    assert 'filter_categories' not in render_args.keys()
+    assert max_frequency >= 0
+
+    # Compute object class frequencies.
+    object_freqs = defaultdict(lambda: 0)
+    for object_ann in nuim.object_ann:
+        category = nuim.get('category', object_ann['category_token'])
+        object_freqs[category['name']] += 1
+
+    # Find rare classes.
+    total_freqs = len(nuim.object_ann)
+    filter_categories_freq = sorted([k for (k, v) in object_freqs.items() if v / total_freqs <= max_frequency])
```

Comment:
Is it necessary to sort? Since only the classes that are rare are in the list

Reply:
Purely for the print statement below to be in order